packages/wallet/src/redux/protocols/transaction-submission/actions.ts

```diff
@@ -1,52 +1,60 @@
+import { WalletProtocol } from '../../types';
+import { BaseProcessAction } from '../actions';
+
 export type TransactionAction =
   | TransactionSentToMetamask
   | TransactionSubmissionFailed
   | TransactionSubmitted
   | TransactionConfirmed
   | RetryTransaction;
 
-export const TRANSACTION_SENT_TO_METAMASK = 'WALLET.COMMON.TRANSACTION_SENT_TO_METAMASK';
-export const TRANSACTION_SUBMISSION_FAILED = 'WALLET.COMMON.TRANSACTION_SUBMISSION_FAILED';
-export const TRANSACTION_SUBMITTED = 'WALLET.COMMON.TRANSACTION_SUBMITTED';
-export const TRANSACTION_CONFIRMED = 'WALLET.COMMON.TRANSACTION_CONFIRMED';
-export const RETRY_TRANSACTION = 'WALLET.COMMON.RETRY_TRANSACTION';
+export const TRANSACTION_SENT_TO_METAMASK = 'WALLET.TRANSACTION_SENT_TO_METAMASK';
+export const TRANSACTION_SUBMISSION_FAILED = 'WALLET.TRANSACTION_SUBMISSION_FAILED';
+export const TRANSACTION_SUBMITTED = 'WALLET.TRANSACTION_SUBMITTED';
+export const TRANSACTION_CONFIRMED = 'WALLET.TRANSACTION_CONFIRMED';
+export const RETRY_TRANSACTION = 'WALLET.RETRY_TRANSACTION';
+export const TRANSACTION_FINALIZED = 'WALLET.TRANSACTION_FINALIZED';
 
-export interface TransactionSentToMetamask {
+export interface TransactionSentToMetamask extends BaseProcessAction {
   type: typeof TRANSACTION_SENT_TO_METAMASK;
+  protocol: WalletProtocol.TransactionSubmission;
```

Comment:
I'm confused as to why these actions have both a `protocol` and a `processId`. In any case, it seems that the `protocol` here can be deduced from the action type. Maybe I don't understand how this `protocol` property will be used.

Reply:
I believe this is from a commit that I cherry-picked from your branch, and I opted not to remove the `protocol` property quite yet. In fact, the proposal suggests _never_ having a `protocol` and `processId` property, which these interfaces have.

Reply:
Not being sure where the `protocol` came from, I've removed it.